cosmos/providers/dbt/core/profiles/base.py

```diff
@@ -0,0 +1,167 @@
+"""
+This module contains a base class that other profile mappings should
+inherit from to ensure consistency.
+"""
+from __future__ import annotations
+
+from logging import getLogger
+from typing import Any
+
+import yaml
+from typing_extensions import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from airflow.models import Connection
+
+logger = getLogger(__name__)
+
+
+class BaseProfileMapping:
```

Comment:
WDYT if we made this an [abstract class](https://docs.python.org/3/library/abc.html) inheriting from `abc.ABC`?